packages/front-end/components/Features/FeatureDiagnostics.tsx

```diff
@@ -0,0 +1,314 @@
+import Link from "@/ui/Link";
+import Tooltip from "@/ui/Tooltip";
+import EmptyState from "@/components/EmptyState";
+
+type FeatureEvaluationDiagnosticsQueryResults = {
+  rows?: FeatureEvalDiagnosticsQueryResponseRows;
+  statistics?: QueryStatistics;
+  error?: string;
+  sql?: string;
+  truncated?: boolean;
+};
+
+// Helper function to format a value for display
+const formatDisplayValue = (value: unknown): string => {
+  if (value === null) {
+    return "";
+  } else if (value === undefined) {
+    return "";
+  } else if (typeof value === "boolean") {
+    return String(value);
+  } else if (typeof value === "object") {
+    return JSON.stringify(value);
+  } else {
+    return String(value);
+  }
+};
+
+export function getFeatureDatasourceDefaults(
+  datasources: DataSourceInterfaceWithParams[],
+  settings: OrganizationSettings,
+  project?: string,
+): { datasourceId: string } {
+  const validDatasources = datasources.filter((d) =>
+    isProjectListValidForProject(d.projects, project),
+  );
+
+  if (!validDatasources.length) return { datasourceId: "" };
+
+  // Default to the first datasource with a feature usage query
+  // If no datasource with a feature usage query is found, default to the default datasource
+  const initialId =
+    validDatasources.find(
+      (d) =>
+        d.settings.queries?.featureUsage &&
+        d.settings.queries?.featureUsage.length > 0,
+    )?.id || settings.defaultDataSource;
+
+  const initialDatasource =
+    (initialId && validDatasources.find((d) => d.id === initialId)) ||
+    validDatasources[0];
+
+  return {
+    datasourceId: initialDatasource.id,
+  };
+}
+
+export default function FeatureDiagnostics({
+  feature,
+  results,
+  setResults,
+}: {
+  feature: FeatureInterface;
+  results: Array<
+    FeatureEvalDiagnosticsQueryResponseRows[number] & { id: string }
+  > | null;
+  setResults: (
+    results: Array<
+      FeatureEvalDiagnosticsQueryResponseRows[number] & { id: string }
+    >,
+  ) => void;
+}) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const { datasources, getDatasourceById } = useDefinitions();
+  const settings = useOrgSettings();
+  const { apiCall } = useAuth();
+
+  const validDatasources = useMemo(() => {
+    return datasources.filter((d) =>
+      isProjectListValidForProject(d.projects, feature.project),
+    );
+  }, [datasources, feature.project]);
+
+  const form = useForm({
+    defaultValues: {
+      ...getFeatureDatasourceDefaults(datasources, settings, feature.project),
+    },
+  });
+
+  const datasourceId = form.watch("datasourceId");
+  const datasource = datasourceId ? getDatasourceById(datasourceId) : null;
+
+  const datasourceHasFeatureUsageQuery =
+    datasource &&
+    datasource.settings.queries?.featureUsage &&
+    datasource.settings.queries.featureUsage.length > 0;
+
+  // Extract all unique keys from results (excluding 'id' which is added internally)
```

Comment:
We should update this comment or remove the specification of the id column given that we are actually removing 3 columns.